internal/component/database_observability/postgres/collector/schema_table.go

```diff
@@ -0,0 +1,340 @@
+package collector
+
+import (
+	"context"
+	"database/sql"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/go-kit/log"
+	"go.uber.org/atomic"
+
+	"github.com/grafana/alloy/internal/component/common/loki"
+	"github.com/grafana/alloy/internal/component/database_observability"
+	"github.com/grafana/alloy/internal/runtime/logging"
+	"github.com/grafana/alloy/internal/runtime/logging/level"
+)
+
+const (
+	OP_SCHEMA_DETECTION = "schema_detection"
+	OP_TABLE_DETECTION  = "table_detection"
+	OP_CREATE_STATEMENT = "create_statement"
+	SchemaTableName     = "schema_table"
+)
+
+const (
+	selectDatabaseName = `SELECT current_database()`
+
+	selectSchemaNames = `
+	SELECT 
+	    nspname as schema_name
+	FROM 
+	    pg_catalog.pg_namespace
+	WHERE 
+	    nspname NOT IN ('information_schema', 'pg_catalog', 'pg_toast')
+	    AND nspname NOT LIKE 'pg_temp_%'
+	    AND nspname NOT LIKE 'pg_toast_%'`
+
+	selectTableNames = `
+	SELECT 
+		pg_class.relname as table_name
+	FROM pg_catalog.pg_class pg_class
+	JOIN pg_catalog.pg_namespace pg_namespace ON pg_class.relnamespace = pg_namespace.oid
+	WHERE pg_namespace.nspname = $1 
+		AND pg_class.relkind IN ('r', 'v', 'm', 'f', 'p')  -- filter for application-facing objects
```

Comment:
nit: here and in the other queries, I'd use code comments rather than sql comments (because the latter are uselessly sent to the database)